### DISCUSSION TITLE
tutorials/lift-off-part4/01-feature-overview

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#58] [course feedback]
# tutorials/lift-off-part4/01-feature-overview

https://www.apollographql.com/tutorials/lift-off-part4/01-feature-overview

**Comment by lst2018** [comment]
thank you very much, nice course, finally done it
interested, it is not easy to apply this to my work but i think the more i learn would make it possible

**Comment by Sajanjohal49** [comment]
In TypeScript projects, if you encounter any errors on the client side related to gql after running 'npm install && npm start,' you can try running 'npm run generate' to resolve the issue. Hope this helps!

**Reply by lizhennessy** [reply]
Thanks so much for this helpful tip for other Odyssey learners, @Sajanjohal49 ! I will make sure to add this to our course as a troubleshooting note for anyone else who might encounter problems.

**Comment by mihaichildesco** [comment]
How do i deploy the app on netlify ? do i need some special configutaion for deploy ? I uploaded to git and from netlify i upoaded the project but it doesnt work

**Reply by lizhennessy** [reply]
Hi @mihaichildesco ! In our case, we split up the `server` and `client` directories and deployed them separately, as this comes closest to emulating how a real production client/server communication process would occur. For the client application, we used Netlify (here's [a helpful guide](https://www.netlify.com/blog/2016/09/29/a-step-by-step-guide-deploying-on-netlify/) I found on the subject). For the server, we used Heroku which [can also integrate with a GitHub repo](https://devcenter.heroku.com/articles/github-integration). By splitting them up, we can let each service handle just one running process. I hope this helps! Let us know if you have any other questions.
